catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -662,46 +662,40 @@ private Metacard generateMetacard(String mimeTypeRaw, String id, String fileName
             throws MetacardCreationException, MimeTypeParseException {
 
         Metacard generatedMetacard = null;
-        InputTransformer transformer = null;
-        StringBuilder causeMessage = new StringBuilder("Could not create metacard with mimeType ");
-        try {
-            MimeType mimeType = new MimeType(mimeTypeRaw);
+        InputTransformer transformer;
+        StringBuilder causeMessage = new StringBuilder();
 
-            List<InputTransformer> listOfCandidates =
-                    frameworkProperties.getMimeTypeToTransformerMapper()
-                            .findMatches(InputTransformer.class, mimeType);
+        MimeType mimeType = new MimeType(mimeTypeRaw);
 
-            LOGGER.debug("List of matches for mimeType [{}]: {}", mimeType, listOfCandidates);
+        List<InputTransformer> listOfCandidates = frameworkProperties.getMimeTypeToTransformerMapper().findMatches(InputTransformer.class, mimeType);
 
-            for (InputTransformer candidate : listOfCandidates) {
-                transformer = candidate;
+        LOGGER.debug("List of matches for mimeType [{}]: {}", mimeType, listOfCandidates);
 
-                try (InputStream transformerStream = com.google.common.io.Files.asByteSource(
-                        tmpContentPath.toFile())
-                        .openStream()) {
-                    generatedMetacard = transformer.transform(transformerStream);
-                }
-                if (generatedMetacard != null) {
-                    break;
+        for (InputTransformer candidate : listOfCandidates) {
+            transformer = candidate;
+
+            try (InputStream transformerStream = com.google.common.io.Files.asByteSource(tmpContentPath.toFile()).openStream()) {
+                generatedMetacard = transformer.transform(transformerStream);
+            } catch (CatalogTransformerException | IOException e) {
+                causeMessage.append(String.format("Transformer [%s] could not create metacard.", transformer));
+                /*
+                    The caught exception more than likely does not have the root cause message
```

Comment:
I think you might be able to replace most of this code with a call to `ExceptionUtils` from commons-lang.

https://commons.apache.org/proper/commons-lang/javadocs/api-3.5/org/apache/commons/lang3/exception/ExceptionUtils.html#getRootCauseStackTrace-java.lang.Throwable-